client/reader/stream/post.jsx

```diff
@@ -195,7 +195,14 @@ const Post = React.createClass( {
 		let postToOpen = this.props.post;
 		// For Discover posts (but not site picks), open the original post in full post view
 		if ( this.state.originalPost ) {
-			postToOpen = this.state.originalPost;
+			postToOpen = { ...this.state.originalPost };
```

Comment:
this could just be 
```
postToOpen = {
  ...this.state.originalPost, 
  { 
    referral: {
      blogId: DiscoverHelper.discoverBlogId,
      postId: this.props.ID
    }
}
```